openapi/src/main/scala/endpoints/openapi/OpenApi.scala

```diff
@@ -0,0 +1,131 @@
+package endpoints.openapi
+
+import io.circe.{Json, JsonObject, ObjectEncoder}
+
+/**
+  * @see [[https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/3.0.md]]
+  */
+case class OpenApi(
+  info: Info,
+  paths: Map[String, PathItem]
+)
+
+object OpenApi {
+
+  implicit val jsonEncoder: ObjectEncoder[OpenApi] =
+    ObjectEncoder.instance { openApi =>
+
+      def jsonMediaTypes(mediaTypes: Map[String, MediaType]): Json =
+        Json.fromFields(mediaTypes.map { case (tpe, mediaType) => tpe -> Json.obj() /* TODO */ })
```

Comment:
Here, for each media type supported by the endpoint we can define a corresponding schema or entity example (see the [spec](https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/3.0.md#mediaTypeObject))